x/oracle/keeper/keeper.go

```diff
@@ -80,10 +89,33 @@ func (k Keeper) GetExchangeRate(ctx sdk.Context, denom string) (sdk.Dec, error)
 		return sdk.ZeroDec(), sdkerrors.Wrap(types.ErrUnknownDenom, denom)
 	}
 
-	dp := sdk.DecProto{}
-	k.cdc.MustUnmarshal(b, &dp)
+	decProto := sdk.DecProto{}
+	k.cdc.MustUnmarshal(b, &decProto)
 
-	return dp.Dec, nil
+	return decProto.Dec, nil
+}
+
+// GetExchangeRateBase gets the consensus exchange rate of an asset
+// in the base denom (e.g. ATOM -> uatom)
+func (k Keeper) GetExchangeRateBase(ctx sdk.Context, denom string) (sdk.Dec, error) {
+	store := ctx.KVStore(k.storeKey)
+	b := store.Get(types.GetExchangeRateKey(denom))
+	if b == nil {
+		return sdk.ZeroDec(), sdkerrors.Wrap(types.ErrUnknownDenom, denom)
+	}
+
+	decProto := sdk.DecProto{}
+	k.cdc.MustUnmarshal(b, &decProto)
```

Comment:
This still is technically not correct as the `denom` input is the "base" not the "symbol". However, `GetExchangeRate` already does all the work for us! So we can make this much simpler:

```suggestion
	exchangeRate, err := k.GetExchangeRate(ctx, denom)
	if err != nil {
	    // ...
	}
```

Now we have the exchange rate for the symbol that corresponds to the input `denom`. We can then use this for the remainder of the method 👍